### ISSUE TITLE
Problem about Reimplementation!!!

### ISSUE THREAD

**Issue by lubinBoooos** [RunpeiDong/PointDistiller#12]
Dear author,
 I try ur code to reimplement the performance of KD, but I cannot get same result as ur paper mentioned. Can u provide some training log about the best performance?

**Comment by RunpeiDong** [comment]
Hi @lubinBoooos,

It's [here](https://drive.google.com/drive/folders/1jA14eMk-0fIywFxku-ijfMIBzFTp7Lef).

**Comment by lubinBoooos** [comment]
Hi @RunpeiDong,
I followed ur log, while training, I found that my KD loss decreasing faster, and I cannot find the kd_cfg setting in ur log, so I want to know ur hyperparameters:  num_voxels and kneighbours settings. Currently, I set it by num_voxels=6000, kneighbours=128.

**Comment by RunpeiDong** [comment]
Which model and which dataset?

**Comment by lubinBoooos** [comment]
**pointpillar 16x  on KITTI dataset**, I got result:
Overall AP40@easy, moderate, hard:
bbox AP40:72.5636, 64.4526, 61.3323
bev  AP40:70.5804, 61.2102, 57.6306
3d   AP40:63.7857, 52.1458, 48.4905
aos  AP40:61.45, 53.31, 50.35

 **while ur log ：**
Overall AP40@easy, moderate, hard:
bbox AP40:76.0208, 66.9024, 63.6359
bev  AP40:73.5277, 63.4987, 59.6674
3d   AP40:67.2419, 54.9371, 50.9578
aos  AP40:65.59, 56.35, 53.32

**Here is my config:**
sys.platform: linux
Python: 3.8.8 (default, Apr 13 2021, 19:58:26) [GCC 7.3.0]
CUDA available: True
GPU 0,1,2,3,4,5,6,7: NVIDIA GeForce RTX 3090
CUDA_HOME: /usr/local/cuda-11.3
NVCC: Build cuda_11.3.r11.3/compiler.29745058_0
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
PyTorch: 1.10.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.2
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.11.0
OpenCV: 4.9.0
MMCV: 1.4.8
MMCV Compiler: GCC 9.4
MMCV CUDA Compiler: 11.3
MMDetection: 2.22.0
MMSegmentation: 0.22.1
MMDetection3D: 1.0.0rc0+a5dc465

**Comment by RunpeiDong** [comment]
Hi @lubinBoooos,

I am reviewing my history of unsolved git issues, and I find that I haven't answered your question. Do you still need help? It's been a long time, but I am happy to provide any help if needed.
